operator/backrest/restore.go

```diff
@@ -143,21 +140,14 @@ func Restore(restclient *rest.RESTClient, namespace string, clientset *kubernete
 	}
 
 	// set up a map of the names of the tablespaces as well as the storage classes
-	// to use for them
-	//
-	// also use it as an opportunity to create the new PVCs for each tablespace
-	// if there is an error at any step in the process, return
-	tablespaceMountsMap := map[string]string{}
-	for tablespaceName, storageSpec := range cluster.Spec.TablespaceMounts {
-		tablespaceMountsMap[tablespaceName] = storageSpec.StorageType
-
-		// get the tablespace PVC name!
-		tablespacePVCName := operator.GetTablespacePVCName(pvcName, tablespaceName)
+	tablespaceStorageTypeMap := operator.GetTablespaceStorageTypeMap(cluster.Spec.TablespaceMounts)
 
-		// attempt to create the PVC
-		if err := createPVC(clientset, restclient, namespace, clusterName, tablespacePVCName, storageSpec); err != nil {
-			log.Error(err)
-			return
+	// combine supplemental groups from all volumes
+	var supplementalGroups []int64
+	{
+		supplementalGroups = append(supplementalGroups, dataVolume.SupplementalGroups...)
+		for _, v := range tablespaceVolumes {
+			supplementalGroups = append(supplementalGroups, v.SupplementalGroups...)
 		}
 	}
```

Comment:
Not totally sure how I feel about the outer `{ }` here

Reply:
These brackets can be removed. In an earlier (unsubmitted) version, there was more going on inside.